packets.go

```diff
@@ -632,6 +632,16 @@ func (rows *textRows) readRow(dest []driver.Value) error {
 							string(dest[i].([]byte)),
 							mc.cfg.loc,
 						)
+
+						if err != nil && mc.invalidTimeAsZero {
+							//in some mysql configurations - possible to store dates in not valid format,
+							//so we just will interpretate them as zero-date
+							if _, ok := err.(*time.ParseError); ok {
+								dest[i] = time.Time{}
+								err = nil
+							}
+						}
+
 						if err == nil {
 							continue
 						}
```

Comment:
Please move this check here (`} else if mc.invalidTimeAsZero {`...) and add an `continue` statement at the end of the block.